modules/react-map-gl-draw/src/editor.js

```diff
@@ -400,13 +398,19 @@ export default class Editor extends PureComponent<EditorProps, EditorState> {
           this._isLine(elem)
         ) {
           const [, , segmentId] = elem.id.split('.');
+          // segmentId is start vertexIndex
+          const vertexIndex = Number(segmentId);
```

Comment:
since you are using Number to make sure segmentId is an actual number, what would you do if the conversation fails (result is NaN)?

Reply:
One suggestion I have to better handling segmentId, you can use the __data__ html property https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes